src/cacheout/cache.py

```diff
@@ -340,6 +353,9 @@ def _set(self, key: t.Hashable, value: t.Any, ttl: t.Optional[T_TTL] = None) ->
         if ttl and ttl > 0:
             self._expire_times[key] = self.timer() + ttl
 
+        if self._enable_stats:
+            self._stats_tracker._total_count = len(self)
```

Comment:
Wondering if instead of doing this, we pass the `Cache` instance to `StatsTracker` and then do `return Stats(..., total_entries=len(self._cache))` so that the total is always accurate and we don't have to worry about syncing up cache key addition/removal events.

Reply:
It will lead to a circular import if `StatsTracker` holds the `Cache` instance.

Reply:
> circular import

Because of cross-typing? That can be bypassed by checking `typing.TYPE_CHECKING` and only importing when `True` in stats module.

This blog as a simple example: https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/

Reply:
Got it! Thanks.